packages/hdwallet-tallyho/src/tallyho.ts

```diff
@@ -62,6 +63,53 @@ export class TallyHoHDWalletInfo implements core.HDWalletInfo, core.ETHWalletInf
     return chainId === 1;
   }
 
+  private async detectTallyProvider(): Promise<TallyHoEthereumProvider | null> {
+    let handled = false;
+
+    return new Promise((resolve) => {
+      if ((window as Window).ethereum) {
+        // eslint-disable-next-line @typescript-eslint/no-use-before-define
+        handleEthereum();
+      } else {
+        // eslint-disable-next-line @typescript-eslint/no-use-before-define
+        window.addEventListener("ethereum#initialized", handleEthereum, { once: true });
+
+        setTimeout(() => {
+          // eslint-disable-next-line @typescript-eslint/no-use-before-define
+          handleEthereum();
+        }, 3000);
+      }
+
+      function handleEthereum() {
+        if (handled) {
+          return;
+        }
+        handled = true;
+
+        window.removeEventListener("ethereum#initialized", handleEthereum);
+
+        const { ethereum } = window as Window;
+
+        if (ethereum && ethereum.isTally) {
+          resolve(ethereum as unknown as TallyHoEthereumProvider);
+        } else {
+          const message = ethereum ? "Non-TallyHo window.ethereum detected." : "Unable to detect window.ethereum.";
+
+          console.error("hdwallet-tallyho: ", message);
+          resolve(null);
+        }
+      }
+    });
+  }
+
+  public async ethGetChainId(): Promise<number> {
+    // at this point, we know that we're in the context of a valid TallyHo provider
+    const provider: any = await this.detectTallyProvider();
+    // chainId as hex string
+    const chainId: string = await provider.request({ method: "eth_chainId" });
+    return parseInt(chainId, 16);
+  }
+
```

Comment:
I think we only need these methods in the `TalloHoHDWallet` class.

Reply:
Seems to work without, removed in a53319d3ad68006635a6bba6dd5d8079c718d32d